lib/active_merchant/billing/gateways/checkout_v2.rb

```diff
@@ -508,7 +509,7 @@ def response(action, succeeded, response, options = {}, source_id = nil)
       end
 
       def headers(action, options)
-        auth_token = @access_token ? "Bearer #{@access_token}" : @options[:secret_key]
+        auth_token = "Bearer #{@options[:access_token] || @options[:secret_key]}"
```

Comment:
This could be a breaking change as this is altering the authentication from `Basic` to `Bearer` auth for gateway tokens that only have a secret_key. Why are we making this change? Is it compatible to use Bearer auth with a secret_key?

Reply:
This change is because the following [guide](https://www.checkout.com/docs/previous/upgrade-to-our-new-platform/payments/test-new-platform-payments/full-card-api) and [this ticket](https://spreedly.atlassian.net/browse/ECS-3487) that described it, to continue supporting the `secret_key` in current merchants, I could add a condition that when the `secret_key` includes `sk_sbox_` add `Bearer` and otherwise do not include the `Bearer` way

Reply:
I have an assumption that the previous `secret_key` ends with a uuid v4, so I am going to upload my changes so that all the tests in CORE work and once I have confirmation if my assumption is correct or not I will update the code (if it is necessary) and mention it to continue with the review :)

Reply:
I removed these changes from this PR, they caused a lot of confusion and they will be in a new PR